src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -34,12 +34,12 @@ public User register(String email, String password) {
     }
 
     @Override
-    public User login(String login, String password) throws AuthenticationException {
-        Optional<User> user = userService.findByEmail(login);
-        String encodedPassword = passwordEncoder.encode(password);
-        if (user.isEmpty() || !user.get().getPassword().equals(encodedPassword)) {
-            throw new AuthenticationException("Incorrect username or password!!!");
+    public User login(String login, String password) {
+        User user = userService.findByEmail(login).orElseThrow(NoSuchElementException::new);
+        if (!passwordEncoder.matches(password, user.getPassword())) {
+            throw new InvalidJwtAuthenticationException("Wrong password or login",
+                    new RuntimeException());
```

Comment:
I'd rework this part to make method throw same InvalidJwtAuthenticationException in both cases (if login is not found and if password is incorrect)